packages/location_permissions/android/src/main/java/com/baseflow/location_permissions/LocationPermissionsPlugin.java

```diff
@@ -169,12 +168,20 @@ public void onMethodCall(MethodCall call, Result result) {
 
   @Override
   public void onListen(Object arguments, EventSink events) {
-    mEventSink = events;
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+      getActiveContext().registerReceiver(mReceiver, mIntentFilter);
+      mEventSink = events;
+    } else {
+      throw new UnsupportedOperationException("Location service availability stream requires at least Android K.");
+    }
   }
 
   @Override
   public void onCancel(Object arguments) {
-    mEventSink = null;
+    if (mEventSink != null) {
+      getActiveContext().unregisterReceiver(mReceiver);
```

Comment:
And here make sure we unsubscribe from the same context:

```
if (mEventSink != null && mContext != null) {
  mContext.unregisterReceiver(mReceiver);
  mContext = null;
  mEventSink = null;
}
```

Reply:
Resolved by agreeing not to store the context?